syft/frameworks/torch/mpc/primitives.py

```diff
@@ -264,3 +265,34 @@ def build_separate_triples(n_party: int, n_instances: int, **kwargs) -> list:
             return primitives_worker
 
         return build_separate_triples
+
+    def __get_next_elem(self, generator, shape=(1,), size=2 ** 32):
+        tensor = th.empty(shape, dtype=th.long)
+        worker = tensor.owner
+
+        return tensor.random_(-size // 2, (size - 1) // 2, generator=generator)
+
+    def generate_alpha_3of3(self):
+        assert (
+            "przs" in self.generators
+        ), "You must call setup_przs because the seeds where not shared between workers"
+
+        generators = self.generators["przs"]
```

Comment:
isn't it supposed to be current and next? every party should hold seed i and seed i+1 I believe

Reply:
That is for the share component - at [Notation](https://arxiv.org/pdf/2004.02229.pdf) you can see that party ```i``` holds ```x_i``` and ```x_i+1```. For RNG they are sending the seed to the next party